### ISSUE TITLE
[Bug] Fenix crashes when TalkBack is disabled

### ISSUE THREAD

**Issue by eeejay** [mozilla-mobile/fenix#2694]
## Steps to reproduce
1. Open Fenix with TalkBack enabled
1. Go to a website.
1. Press tabs button to go back to home screen.
1. Disable TalkBack with shortcut (press both volume keys)

### Expected behavior
TalkBack should shut off, and Fenix should not crash.

### Actual behavior
Fenix crashes:

```2
    Process: org.mozilla.fenix.debug, PID: 6620
    java.lang.IllegalStateException: Fragment BrowserFragment{51f31c (b37efd4f-13fc-49c8-8747-0db12183660c)} not attached to a context.
        at androidx.fragment.app.Fragment.requireContext(Fragment.java:763)
        at org.mozilla.fenix.ext.FragmentKt.getRequireComponents(Fragment.kt:13)
        at org.mozilla.fenix.browser.BrowserFragment.getSessionById(BrowserFragment.kt:716)
        at org.mozilla.fenix.browser.BrowserFragment.updateToolbar(BrowserFragment.kt:751)
        at org.mozilla.fenix.browser.BrowserFragment.access$updateToolbar(BrowserFragment.kt:96)
        at org.mozilla.fenix.browser.BrowserFragment$onStart$1.onTouchExplorationStateChanged(BrowserFragment.kt:396)
        at android.view.accessibility.AccessibilityManager.lambda$notifyTouchExplorationStateChanged$1(AccessibilityManager.java:1174)
        at android.view.accessibility.-$$Lambda$AccessibilityManager$a0OtrjOl35tiW2vwyvAmY6_LiLI.run(Unknown Source:4)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

### Device information
* Android device: Pixel
* Fenix version: master (`9c2352f`)

**Comment by kbrosnan** [comment]
Sentry issue: [FENIX-17T](https://sentry.prod.mozaws.net/operations/fenix/issues/5790542/)

New crash in release 1.0.1921.

**Comment by ekager** [comment]
Should be temporarily fixed by removing the accessibility listener (because we are using the fixed toolbar for now) https://github.com/mozilla-mobile/fenix/pull/2742

**Comment by ekager** [comment]
For reference when we reimplement the dynamic toolbar the root cause of this was I don't think I was unregistering the listener correctly.
